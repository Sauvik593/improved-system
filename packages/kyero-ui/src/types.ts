import type { LinkHTMLAttributes } from 'react';

export type BaseLinkProps =
  | {
      to: string;
    }
  | {
      to: {
        search?: string;
        pathname?: string;
      };
    };

export type LinkProps = BaseLinkProps & LinkHTMLAttributes<HTMLAnchorElement>;

export type LinkComponentProps = {
  className: string;
  children: React.ReactNode;
} & LinkProps;
