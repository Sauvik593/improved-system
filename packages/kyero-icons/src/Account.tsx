export const Account = () => {
  return (
    <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        d="M5.625 8.125C5.625 9.28532 6.08594 10.3981 6.90641 11.2186C7.72688 12.0391 8.83968 12.5 10 12.5C11.1603 12.5 12.2731 12.0391 13.0936 11.2186C13.9141 10.3981 14.375 9.28532 14.375 8.125C14.375 6.96468 13.9141 5.85188 13.0936 5.03141C12.2731 4.21094 11.1603 3.75 10 3.75C8.83968 3.75 7.72688 4.21094 6.90641 5.03141C6.08594 5.85188 5.625 6.96468 5.625 8.125V8.125Z"
        stroke="#1F4DEF"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M15.7609 17.3968C15.0061 16.6372 14.1085 16.0344 13.1199 15.6231C12.1312 15.2118 11.0709 15 10.0001 15C8.92926 15 7.869 15.2118 6.88032 15.6231C5.89165 16.0344 4.99408 16.6372 4.23926 17.3968"
        stroke="#1F4DEF"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M0.625 10C0.625 12.4864 1.61272 14.871 3.37087 16.6291C5.12903 18.3873 7.5136 19.375 10 19.375C12.4864 19.375 14.871 18.3873 16.6291 16.6291C18.3873 14.871 19.375 12.4864 19.375 10C19.375 7.5136 18.3873 5.12903 16.6291 3.37087C14.871 1.61272 12.4864 0.625 10 0.625C7.5136 0.625 5.12903 1.61272 3.37087 3.37087C1.61272 5.12903 0.625 7.5136 0.625 10V10Z"
        stroke="#1F4DEF"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  );
};
