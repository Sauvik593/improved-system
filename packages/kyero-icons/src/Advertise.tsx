export const Advertise = () => {
  return (
    <svg
      width="20"
      height="20"
      viewBox="0 0 20 20"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      aria-hidden="true"
    >
      <path
        d="M8.93115 12.8125C7.68115 11.5625 7.68115 10.2125 8.93115 8.98126C10.1812 7.75001 11.5312 7.73126 12.7624 8.98126"
        stroke="white"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      ></path>
      <path
        d="M8.93115 7.70605L11.4874 10.2561"
        stroke="white"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      ></path>
      <path
        d="M7.65625 8.98096L9.575 10.8935"
        stroke="white"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      ></path>
      <path
        d="M13.9 6.80635C14.3971 6.80635 14.8 6.4034 14.8 5.90635C14.8 5.40929 14.3971 5.00635 13.9 5.00635C13.4029 5.00635 13 5.40929 13 5.90635C13 6.4034 13.4029 6.80635 13.9 6.80635Z"
        stroke="white"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      ></path>
      <path
        d="M16.0625 2.96877H10.2062C10.0739 2.96957 9.94319 2.99712 9.82182 3.04975C9.70045 3.10238 9.59098 3.179 9.49998 3.27502L3.24998 9.94377C3.16106 10.0376 3.09209 10.1485 3.04723 10.2697C3.00237 10.3909 2.98255 10.52 2.98898 10.6491C2.9954 10.7782 3.02794 10.9047 3.08461 11.0208C3.14129 11.137 3.22093 11.2405 3.31873 11.325L9.56873 16.7938C9.75799 16.9586 10.0038 17.0436 10.2545 17.0307C10.5051 17.0179 10.741 16.9083 10.9125 16.725L16.7687 10.475C16.9374 10.2982 17.0314 10.0631 17.0312 9.81877V3.93127C17.0312 3.80435 17.0061 3.67867 16.9574 3.56149C16.9086 3.4443 16.8372 3.33792 16.7471 3.24846C16.6571 3.159 16.5502 3.08824 16.4327 3.04023C16.3152 2.99223 16.1894 2.96795 16.0625 2.96877V2.96877Z"
        stroke="white"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      ></path>
    </svg>
  );
};
