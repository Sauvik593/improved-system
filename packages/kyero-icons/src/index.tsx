export * from './AnimatedSpinner';
export * from './ArrowUp';
export * from './Account';
export * from './ArrowDown';
export * from './Advertise';
export * from './ArrowBackward';
export * from './ArrowForward';
export * from './AlertInfo';
export * from './Avatar';
export * from './BoostRocket';
export * from './Camera';
export * from './Cart';
export * from './Close';
export * from './Chat';
export * from './Checked';
export * from './ChevronDown';
export * from './ChevronLeft';
export * from './Dashboard';
export * from './DeleteIcon';
export * from './Edit';
export * from './Email';
export * from './Eye';
export * from './Floorplans';
export * from './Forbidden';
export * from './Furniture';
export * from './Gate';
export * from './Globe';
export * from './Golf';
export * from './Hamburger';
export * from './HouseIcon';
export * from './JoinLogo';
export * from './List';
export * from './Logout';
export * from './LogoWide';
export * from './Imports';
export * from './InfoIcon';
export * from './Kyero';
export * from './NextIcon';
export * from './Outside';
export * from './Parking';
export * from './Photo';
export * from './Plus';
export * from './Pool';
export * from './PrevIcon';
export * from './Processing';
export * from './Properties';
export * from './Quote';
export * from './RemoteViewing';
export * from './Reply';
export * from './School';
export * from './Search';
export * from './ShareIcon';
export * from './Star';
export * from './SunMark';
export * from './Temperature';
export * from './Tick';
export * from './Town';
export * from './QuestionMark';
export * from './WavyArrow';
export * from './Views';
export * from './VirtualTour';
export * from './XIcon';
