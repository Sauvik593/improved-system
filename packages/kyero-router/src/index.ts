import { DocsRoutes } from './docs-routes';
import { GuidesRoutes } from './guides-routes';
import { IDRoutes } from './id-routes';
import { JoinRoutes } from './join-routes';
import { SupportRoutes } from './support-routes';
import { DataRoutes } from './data-routes';
import { returnCountryRouteForLocale } from './helpers';

import { preloadRoutesFromAPI } from './preload-routes-from-api';

import type {
  RoutesFromFrontendAPI,
  Countries,
  Locales,
  CountryPreloadedRoute,
  ExternalRoutes,
  FrontendRoutes,
  SocialRoutes,
} from './types';

export const Routes = {
  id: IDRoutes,
  join: JoinRoutes,
  guides: GuidesRoutes,
  docs: DocsRoutes,
  data: DataRoutes,
  support: SupportRoutes,
};

export class KyeroRouter {
  preloadedRoutes: RoutesFromFrontendAPI | null;

  constructor() {
    this.preloadedRoutes = null;
  }

  async preloadRoutes() {
    this.preloadedRoutes = await preloadRoutesFromAPI();
  }

  getExternalRoutes(locale: string) {
    return {
      privacy: Routes.docs.privacyURL(locale),
      terms: Routes.docs.termsURL(locale),
      cookies: Routes.docs.cookiesURL(locale),
      buyersGuide: Routes.guides.buyersGuideURL(locale),
      spainGuide: Routes.guides.spainGuideURL(locale),
      portugalGuide: Routes.guides.portugalGuideURL(locale),
      franceGuide: Routes.guides.franceGuideURL(locale),
      italyGuide: Routes.guides.italyGuideURL(locale),
      pdfGuide: Routes.guides.pdfGuideURL(locale),
      podcasts: Routes.guides.podcastsURL(),
      support: Routes.support.url(),
      agentsFAQ: Routes.support.agentFAQURL(locale),
      generalFAQ: Routes.support.generalFAQURL(locale),
      buyersFAQ: Routes.support.buyersFAQURL(locale),
      join: Routes.join.joinURL(locale),
      joinContact: Routes.join.contactURL(locale),
      marketData: Routes.data.marketData(locale),
      affordability: Routes.data.affordabilityCalculator(locale),
      jobs: 'https://jobs.kyero.com',
      account: Routes.id.settingsURL(locale),
      forgotPassword: Routes.id.passwordResetURL(locale),
    };
  }

  getSocialRoutes() {
    return {
      facebook: 'https://www.facebook.com/kyero.co.uk/',
      twitter: 'https://twitter.com/kyero',
      instagram: 'https://www.instagram.com/kyero_property/',
      linkedin: 'https://www.linkedin.com/company/kyero-com/',
      youtube: 'https://www.youtube.com/channel/UCmgKXDJ0fvVzMAAp4YlMYxA',
      pinterest: 'https://www.pinterest.co.uk/kyeroproperty/',
    };
  }

  getFrontendRoutes(locale: string) {
    const { preloadedRoutes } = this;

    if (!preloadedRoutes) {
      throw new Error('Routes not preloaded');
    }

    return {
      properties_for_sale: returnCountryRouteForLocale(preloadedRoutes.properties_for_sale, locale),
      agents_search: returnCountryRouteForLocale(preloadedRoutes.agents_search, locale),
      properties_to_rent: returnCountryRouteForLocale(preloadedRoutes.properties_to_rent, locale),
      cookie_preferences: preloadedRoutes.cookie_preferences[locale as Locales],
      sitemap: preloadedRoutes.sitemap[locale as Locales],
    };
  }
}

export type {
  Countries,
  Locales,
  CountryPreloadedRoute,
  ExternalRoutes,
  FrontendRoutes,
  SocialRoutes,
};
