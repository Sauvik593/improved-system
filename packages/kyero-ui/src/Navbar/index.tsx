export * from './Navbar';
export * from './NavElement';
