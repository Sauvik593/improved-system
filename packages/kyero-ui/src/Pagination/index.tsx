import cn from 'classnames';
import { Page } from './Page';
import { Button } from './Button';
import { usePagination } from './usePagination';

import type { LinkComponentProps } from '../types';

export interface PaginationProps {
  totalPages: number;
  currentPage: number;
  pathname: string;
  nextPage: number | null;
  prevPage: number | null;
  LinkComponent?: (props: LinkComponentProps) => React.ReactElement;
  search: string;
}

const DefaultComponent = ({ to, children, className }: LinkComponentProps) => {
  const path = typeof to === 'string' ? to : `${to?.pathname}?${to?.search}`;
  return (
    <a href={path} className={cn(className)}>
      {children}
    </a>
  );
};

export const Pagination = ({
  currentPage,
  totalPages,
  pathname,
  prevPage,
  nextPage,
  LinkComponent = DefaultComponent,
  search,
}: PaginationProps) => {
  const { showPrevDots, showNextDots, pagesArray } = usePagination(search, currentPage, totalPages);
  return (
    <nav className="mt-4">
      <ul className="flex items-center gap-2">
        <Button
          page={prevPage}
          pathname={pathname}
          search={search}
          type="prev"
          LinkComponent={LinkComponent}
          testId="pagination-prev"
        />
        {totalPages >= 2 && (
          <Page
            page={1}
            currentPage={currentPage}
            pathname={pathname}
            search={search}
            LinkComponent={LinkComponent}
            testId="pagination-firstpage"
          />
        )}
        {showPrevDots && (
          <li className="font-bold" data-testid="pagination-more-prev-dots">
            ...
          </li>
        )}
        {pagesArray.map((page) => {
          return (
            <Page
              key={page}
              page={page}
              currentPage={currentPage}
              pathname={pathname}
              search={search}
              LinkComponent={LinkComponent}
              testId={`pagination-page-${page}`}
            />
          );
        })}
        {showNextDots && (
          <li className="font-bold" data-testid="pagination-more-next-dots">
            ...
          </li>
        )}
        <Page
          page={totalPages}
          currentPage={currentPage}
          pathname={pathname}
          search={search}
          LinkComponent={LinkComponent}
          testId="pagination-lastpage"
        />
        <Button
          page={nextPage}
          pathname={pathname}
          search={search}
          type="next"
          testId="pagination-next"
          LinkComponent={LinkComponent}
        />
      </ul>
    </nav>
  );
};
