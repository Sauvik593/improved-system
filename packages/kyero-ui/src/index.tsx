export * from './Alert';
export * from './ArrowLink';
export * from './Badge';
export * from './Button';
export * from './Checkbox';
export * from './CheckboxList';
export * from './FiltersField';
export * from './Footer';
export * from './LanguageSwitcher';
export * from './Navbar';
export * from './Pagination';
export * from './Radio';
export * from './RadioList';
export * from './SearchInput';
export * from './Select';
export * from './Switch';
export * from './Tooltip';
