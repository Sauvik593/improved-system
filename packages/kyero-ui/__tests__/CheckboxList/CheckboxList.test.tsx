import renderer from 'react-test-renderer';
import { CheckboxList, type CheckboxListProps } from '../../src';

const OPTIONS = [
  { label: '1', value: '1' },
  { label: '2', value: '2' },
  { label: '3', value: '3' },
];

const getProps = (props: Partial<CheckboxListProps> = {}) => ({
  name: 'Test',
  defaultValue: '1',
  options: OPTIONS,
  ...props,
});

const renderCheckboxList = (props: CheckboxListProps) => <CheckboxList {...props} />;

describe('<CheckboxList />', () => {
  it('should render correctly', () => {
    const props = getProps() as CheckboxListProps;
    const checkboxList = renderer.create(renderCheckboxList(props)).toJSON();
    expect(checkboxList).toMatchSnapshot();
  });
});
