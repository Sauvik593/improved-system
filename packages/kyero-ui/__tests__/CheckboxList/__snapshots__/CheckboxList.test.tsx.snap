// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxList /> should render correctly 1`] = `
<div>
  <label
    className="flex my-2 "
  >
    <input
      checked={true}
      className="relative flex border-sierra-night-40 relative inline-block h-6 w-6 appearance-none overflow-hidden rounded-md border-1 bg-white shadow-sm transition-all after:content[''] after:top-0 after:left-0 after:-right-0 after:bottom-0 after:bg-[url('/images/check.svg')] after:bg-no-repeat after:bg-cover after:w-[80%] after:h-[80%] after:m-auto after:rounded-full after:block after:scale-0 after:transition-all checked:bg-ocean-150 checked:after:scale-100"
      disabled={false}
      name="Test"
      onChange={[Function]}
      type="checkbox"
      value="1"
    />
    <span
      className="text-p-2 ml-3 block w-full"
    >
      1
    </span>
  </label>
  <label
    className="flex my-2"
  >
    <input
      checked={false}
      className="relative flex border-sierra-night-40 relative inline-block h-6 w-6 appearance-none overflow-hidden rounded-md border-1 bg-white shadow-sm transition-all after:content[''] after:top-0 after:left-0 after:-right-0 after:bottom-0 after:bg-[url('/images/check.svg')] after:bg-no-repeat after:bg-cover after:w-[80%] after:h-[80%] after:m-auto after:rounded-full after:block after:scale-0 after:transition-all checked:bg-ocean-150 checked:after:scale-100"
      disabled={false}
      name="Test"
      onChange={[Function]}
      type="checkbox"
      value="2"
    />
    <span
      className="text-p-2 ml-3 block w-full"
    >
      2
    </span>
  </label>
  <label
    className="flex my-2"
  >
    <input
      checked={false}
      className="relative flex border-sierra-night-40 relative inline-block h-6 w-6 appearance-none overflow-hidden rounded-md border-1 bg-white shadow-sm transition-all after:content[''] after:top-0 after:left-0 after:-right-0 after:bottom-0 after:bg-[url('/images/check.svg')] after:bg-no-repeat after:bg-cover after:w-[80%] after:h-[80%] after:m-auto after:rounded-full after:block after:scale-0 after:transition-all checked:bg-ocean-150 checked:after:scale-100"
      disabled={false}
      name="Test"
      onChange={[Function]}
      type="checkbox"
      value="3"
    />
    <span
      className="text-p-2 ml-3 block w-full"
    >
      3
    </span>
  </label>
</div>
`;
